### DISCUSSION TITLE
How to specify prefetch-size client side using JMS?

### DISCUSSION THREAD

**Discussion by tangi11** [rabbitmq/rabbitmq-jms-client#185] [Q&A] [answered]
Hello,
I'd like to specify a "prefetch-size" with a distinct value for each queue.

I can see that RabbitMQ AMQP (https://www.rabbitmq.com/amqp-0-9-1-reference.html) allows specifying a "prefetch-size", and I'd like to do the same while using the RabbitMQ JMS client.
I also found the server-side configuration of "rabbit.default_consumer_prefetch" (https://www.rabbitmq.com/consumer-prefetch.html#default-limit) but I understand it would not be a per-queue configuration.
I also searched for "prefetch" in the source code of rabbitmq-jms-client-2.5.0, but found nothing.

=> What would be the current options compliant with the use the RabbitMQ JMS?

Thank you for your help,
Best regards,
Tangi

**Comment by michaelklishin** [comment, marked as answer]
[Prefetch](https://www.rabbitmq.com/confirms.html) is a channel setting in AMQP 0-9-1, it has no relation to queues.

As of https://github.com/rabbitmq/rabbitmq-jms-client/pull/124 it can be set on connection factory and thus be set per (this client's) connection.

**Reply by tangi11** [reply]
Hello Michael,
Great, everything's clear now!
Indeed I had not the "channel" in mind - JMS encapsulates its use, and I'm a true newbie with RabbitMQ.
Moreover, I could have found the solution by myself if AMQP had not found funny to name the "prefetch" configuration item as "QoS" (which may cover a much wider notion) instead of "prefetch", for instance :-D
Thanks again for your help!
Regards,
Tangi

**Reply by michaelklishin** [reply]
"Prefetch" is discussed in the docs but in the protocol and RabbitMQ's  implementation of AMQP 0-9-1 it is a part of the `basic.qos` operation. I guess we could alias it in supported query properties @acogoluegnes?

**Reply by acogoluegnes** [reply]
There's `RMQConnectionFactory#setChannelsQos(int)` to set up the prefetch size on the channels created by the library. The default is -1, that is not prefetch size set.
